### PR TITLE
Replace 11.1 with 11.2 on CI for Windows

### PR DESCRIPTION
**Pull request by janeyx99**
Adding CUDA 11.2 to Windows CI.

Disabled tests:

The following ran into `CUDA error: misaligned address` for CUDA 11.2: (issue linked below)
`test_where_scalar_valid_combination_cuda_complex128` in test_torch.py
`test_sgn_complex_cuda` in test_autograd.py

The following ran into `CUDA error: too many resources requested for launch` for CUDA 11.2: (https://github.com/pytorch/pytorch/issues/52002)
test_EmbeddingBag_per_sample_weights_and_new_offsets_cuda_int64_float64
test_EmbeddingBag_per_sample_weights_and_offsets_cuda_int64_float64

The following ran into test assertion failures in test_optim.py (https://github.com/pytorch/pytorch/issues/51992)
test_adadelta
test_adam
test_adamw
test_multi_tensor_optimizers
test_rmsprop

### CODE REVIEW

File: .circleci/cimodel/data/windows_build_definitions.py [hunk 1]
```diff
@@ -132,10 +132,10 @@ def TruePred(_):
     WindowsJob(None, _VC2019, CudaVersion(10, 1)),
     WindowsJob(1, _VC2019, CudaVersion(10, 1)),
     WindowsJob(2, _VC2019, CudaVersion(10, 1)),
-    # VS2019 CUDA-11.1
-    WindowsJob(None, _VC2019, CudaVersion(11, 1)),
-    WindowsJob(1, _VC2019, CudaVersion(11, 1), master_only_pred=TruePred),
-    WindowsJob(2, _VC2019, CudaVersion(11, 1), master_only_pred=TruePred),
+    # VS2019 CUDA-11.2
+    WindowsJob(None, _VC2019, CudaVersion(11, 2)),
+    WindowsJob(1, _VC2019, CudaVersion(11, 2), master_only_pred=TruePred),
```
Comment (mszhanyi) [inline .circleci/cimodel/data/windows_build_definitions.py:137]: Please remove master_only_pred=TruePred for windows cuda11.2 first.
If it passed, then restore master_only_pred.

And please paste the link of  windows cuda11.2 test in the PR.

File: .jenkins/pytorch/win-test.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ if [ -n "$CIRCLE_PULL_REQUEST" ]; then
   file_diff_from_base "$DETERMINE_FROM"
 fi
 
-if [[ "${CIRCLE_JOB}" == *11.1* ]]; then
+if [[ "${CIRCLE_JOB}" == *11.2* ]]; then
   export BUILD_SPLIT_CUDA=ON
```
Comment (peterjc123) [inline .jenkins/pytorch/win-test.sh:44]: The test failure seems legit. I wonder whether the test failure is related to this flag.

**Review by peterjc123 — COMMENTED** [review]
The test failure looks legit and doesn't seem to be Windows-only.

**Comment by janeyx99** [comment]
> The test failure looks legit and doesn't seem to be Windows-only.

The windows tests errors don't appear in commits on master, though.

File: .jenkins/pytorch/test.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@ if [[ "$BUILD_ENVIRONMENT" == *coverage* ]]; then
   export PYTORCH_COLLECT_COVERAGE=1
 fi
 
-if [[ "$BUILD_ENVIRONMENT" == *cuda11.1* ]]; then
+if [[ "$BUILD_ENVIRONMENT" == *cuda11* ]]; then
```
Comment (walterddr) [inline .jenkins/pytorch/test.sh:25]: any particular reason why using `cuda11` here and the window side uses `cuda11.2`?

**Comment by mszhanyi** [comment]
@janeyx99 magma for cuda11.2 size is smaller, Is it right?
image

**Comment by peterjc123** [comment]
@mszhanyi The magma binaries are built in this PR: https://github.com/pytorch/pytorch/pull/42408. Looking into the build log, I didn't find anything wrong there.

**Comment by janeyx99** [comment]
> @janeyx99 magma for cuda11.2 size is smaller, Is it right?
> image

Yes, I believe CUDA 11.2 is smaller than CUDA 11.1.

File: .jenkins/pytorch/test.sh (hunk 3 above)
Comment (janeyx99) [reply .jenkins/pytorch/test.sh:25]: No particular reason--I was actually going to change it to just 11 in linux in the linux PR :p

**Review by walterddr — APPROVED** [review]

**Comment by janeyx99** [comment]
After rebasing and rerunning, I don't think the `CUDA error: misaligned address` is separate from CUDA 11.2. @zasdfgbnm @peterjc123 any thoughts?

**Comment by zasdfgbnm** [comment]
@janeyx99 I have never seen this failure. Let me try to reproduce locally.

**Comment by zasdfgbnm** [comment]
I reproduced the failure of `test_where_scalar_valid_combination_cuda_complex128` on Windows, but it is not failing on my Linux system.

**Comment by zasdfgbnm** [comment]
I opened an issue at https://github.com/pytorch/pytorch/issues/51980 for the failing test to track it separately.

File: test/test_nn.py [hunk 4]
```diff
@@ -12685,6 +12685,7 @@ def test_per_sample_weights(mode, trainable_scale):
         for mode, trainable in itertools.product(modes, trainable_scale):
             test_per_sample_weights(mode, trainable)
 
+    @unittest.skipIf(IS_WINDOWS, "FIXME: CUDA error: too many resources requested on 11.2")
```
Comment (zasdfgbnm) [inline test/test_nn.py:12688]: Could we also have a issue for this one, and reference the issue number here?

Comment (janeyx99) [reply test/test_nn.py:12688]: https://github.com/pytorch/pytorch/issues/52002 yes I made one here (linked in description)

**Comment by mszhanyi** [comment]
@zasdfgbnm 
I'm trying to build pytorch with cuda11.2 on windows.
I came across some errors like
```
identifier "__floorf" is undefined in device code, calling a __host__ function("__floorf") from a __device__ function xxx is not allowed.
identifier "__ceilf" is undefined in device code. ....
```
Did you come across them? Is it specific in cuda11.2

**Comment by peterjc123** [comment]
@mszhanyi No, it's caused by the VS upgrade, namely this commit https://github.com/microsoft/STL/pull/1336 in STL. They forget to make the fallback to C_STD function for CUDA device code.

**Pull request closed without merging**